core/tests/protractor_utils/ProfilePage.js

```diff
@@ -70,6 +79,9 @@ var ProfilePage = function() {
   };
 
   this.expectUserToHaveInterestPlaceholder = async function(expectedText) {
+    await waitFor.visibilityOf(
+      interestPlaceholder,
+      'Interest Place Holder is not present or Taking Time To Display');
     expect(await interestPlaceholder.getText()).toMatch(expectedText);
   };
 
```

Comment:
In `expectToHaveExplorationCards` we fail if there are no exploration cards, so we can use a waitFor

Reply:
not done